src/v2/Apps/Conversation/Components/OrderUpdate.tsx

```diff
@@ -55,16 +55,24 @@ export const OrderUpdate: React.FC<OrderUpdateProps> = ({
       return null
     }
   } else if (event.__typename === "CommerceOrderStateChangedEvent") {
-    const { orderUpdateState } = event
-    if (orderUpdateState === "offer_approved") {
+    const { orderUpdateState, state, stateReason } = event
+    const reasonRejected = stateReason?.includes("_rejected")
+    const reasonLapsed = stateReason?.includes("_lapsed")
+    if (state === "APPROVED") {
       color = "green100"
-      message = `Offer Accepted`
-    } else if (orderUpdateState === "offer_rejected") {
+      message = `${
+        orderUpdateState === "offer_approved" ? "Offer" : "Purchase"
+      } Accepted`
+    } else if (state === "CANCELED" && reasonRejected) {
       color = "red100"
-      message = `Offer Declined`
-    } else if (orderUpdateState === "offer_lapsed") {
-      color = "black60"
-      message = `Offer Expired`
+      message = `${
+        orderUpdateState === "offer_rejected" ? "Offer" : "Purchase"
+      } Declined`
```

Comment:
Is `Purchase declined` a state? I thought only offers could be rejected. I would think that the only possibility for a BN order to be canceled/declined would be through expiration (caused by lack of confirmation on the partner side)

Reply:
We've added it to Exchange but I think we don't have the functionality yet (haven't seen the option in CMS) but I thought we might keep it if we add it in the future, or is it better to remove it until then?

Reply:
if the scenario is unachievable, I would think it makes more sense not to add it here as it could be misleading...